src/components/AgeRangeField/AgeRangeInput.tsx

```diff
@@ -0,0 +1,343 @@
+import { Button } from '../Button'
+import { Dropdown, DropdownItem } from '../Dropdown'
+import { FocusManagerContainer } from '../FocusManagerContainer'
+import { Icon } from '../Icon'
+import { Text } from '../Text'
+import { TextInput } from '../TextField'
+import { Tooltip } from '../Tooltip'
+import { convertUnitAgeRangeEnumToLocaleText } from './converter'
+import { AgeRange, AgeRangeUnitEnum } from './model'
+import { getAvaibleAgeRangeUnits } from './util'
+
+interface Placeholders {
+  start?: string
+  end?: string
+}
+
+export interface AgeRangeInputProps {
+  /**
+   * Set a AgeRange as initial value of the component
+   */
+  value?: AgeRange
+
+  /**
+   * Component name
+   */
+  name?: string
+
+  /**
+   * Limit the length of the number
+   */
+  maxLength?: number
+
+  /**
+   * Disable the age range input
+   */
+  disabled?: boolean
+
+  /**
+   * Enable the "x" button to clear value in both of the inputs
+   */
+  clearable?: boolean
+
+  /**
+   *  Reference an invalid state
+   */
+  invalid?: boolean
+
+  /**
+   * Set placeholders in the inputs
+   */
+  placeholders?: Placeholders
+
+  /**
+   *  Receive external styles
+   */
+  style?: ExternalStyles
+
+  /**
+   * Set unit options that won't be shown on drop down menu
+   */
+  unitOptionsToExclude?: AgeRangeUnitEnum[]
+
+  /**
+   * Called whenever the age range changes
+   * @param ageRange
+   */
+  onChange?(ageRange: AgeRange): void
+
+  /**
+   * Called when the start or end input is focused
+   * @param focusEvent
+   */
+  onFocus?(focusEvent: React.FocusEvent<HTMLDivElement>): void
+
+  /**
+   * Called when the start or end input loses focus
+   * @param focusEvent
+   */
+  onBlur?(focusEvent: React.FocusEvent<HTMLDivElement>): void
+}
+
+export function AgeRangeInput(props: AgeRangeInputProps) {
+  const {
+    value,
+    name,
+    maxLength,
+    disabled,
+    clearable,
+    invalid,
+    placeholders,
+    style,
+    unitOptionsToExclude,
+    onChange,
+    onFocus,
+    onBlur,
+  } = props
+
+  const [anchorRef, setAnchorRef] = useState<HTMLButtonElement>()
+  const [open, setOpen] = useState(false)
+
+  const { classes, css } = useStyles(createStyles, disabled)
+  const classNameDiv = css(classes.div, invalid && classes.invalid, style)
+  const locale = useLocale()
+
+  if (unitOptionsToExclude?.includes(value?.unit)) {
+    throw new Error(
+      `You selected ${convertUnitAgeRangeEnumToLocaleText(
+        value.unit,
+        locale
+      )}, but it's an excluded unit option defined in 'unitOptionsToExclude' prop.`
+    )
+  }
+
+  const handleClickUnitOptionsButton = () => setOpen((state) => !state)
+
+  const handleChangeStart = (e: React.ChangeEvent<HTMLInputElement>) => {
+    let start: number = null
+
+    const startText = e.currentTarget?.value
+    const startParsed = parseInt(startText)
+
+    if (!isNaN(startParsed)) {
+      start = startParsed
+    } else if (!isEmpty(startText)) {
+      start = value?.start
+    }
```

Comment:
```suggestion
    const start = parseInt(e.currentTarget?.value) || value?.start
```

Are the isNaN and isEmpty checks really necessary in this case?